### PR TITLE
Igal/upload logs

### PR DESCRIPTION
**Pull request by tsorya**

### CODE REVIEW

File: internal/bminventory/inventory_test.go [hunk 1]
```diff
@@ -1654,8 +1656,132 @@ var _ = Describe("UploadClusterIngressCert test", func() {
...
+		}
+		newHostID := strfmt.UUID(uuid.New().String())
+		host := addHost(newHostID, models.HostRoleMaster, "known", clusterID, "{}", db)
+		fileName := fmt.Sprintf("%s/logs/%s/%s", params.ClusterID, host.RequestedHostname, "test_kubeconfig")
+		mockS3Client.EXPECT().UploadFile(gomock.Any(), gomock.Any(), fileName).Return(nil).Times(1)
+		reply := bm.UploadHostLogs(ctx, params)
+		Expect(reply).Should(BeAssignableToTypeOf(installer.NewUploadHostLogsNoContent()))
+	})
+
+	// TODO add s3 upload testcases after change in s3wrapper
```
Comment (tsorya) [inline internal/bminventory/inventory_test.go:1768]: delete comment, already done

File: internal/bminventory/inventory.go [hunk 2]
```diff
@@ -2035,6 +2035,52 @@ func (b *bareMetalInventory) GetFreeAddresses(ctx context.Context, params instal
...
+		params.ClusterID).Error; err != nil {
+		if gorm.IsRecordNotFoundError(err) {
+			return common.NewApiError(http.StatusNotFound, err)
+		}
+		return common.NewApiError(http.StatusInternalServerError, err)
+	}
+	if len(cluster.Hosts) < 1 {
+		return common.NewApiError(http.StatusNotFound, errors.Errorf("Host %s not found", params.HostID))
+	}
+	// needed to get filename
```
[the remark this answers is not in the document]
Comment (tsorya) [reply internal/bminventory/inventory.go:2064]: are u sure? i don't think service need to think about it. I think we need to return folder with logs and user will check what file it wants.

[the remark this answers is not in the document]
Comment (avishayt) [reply internal/bminventory/inventory.go:2064]: OK

File: internal/bminventory/inventory.go [hunk 3]
```diff
@@ -2035,6 +2035,52 @@ func (b *bareMetalInventory) GetFreeAddresses(ctx context.Context, params instal
...
+	_, fileHeader, err := params.HTTPRequest.FormFile("upfile")
+	if err != nil {
+		log.WithError(err).Errorf("Failed to get filename")
+		return common.NewApiError(http.StatusInternalServerError, err)
+	}
+
+	fileName := fmt.Sprintf("%s/logs/%s/%s", params.ClusterID, cluster.Hosts[0].RequestedHostname, fileHeader.Filename)
+	log.Infof("Start upload %s to bucket %s aws len", fileName, b.S3Bucket)
```
Comment (avishayt) [inline internal/bminventory/inventory.go:2072]: log.Debugf

**Pull request merged**